tests/test_index.py

```diff
@@ -167,3 +167,64 @@ class ExampleIndex(AlgoliaIndex):
 
         with self.assertRaises(AlgoliaIndexError):
             index = ExampleIndex(Example, self.client)
+
+    def test_should_index_method(self):
+        class ExampleIndex(AlgoliaIndex):
+            fields = 'name'
+            should_index = 'has_name'
+        index = ExampleIndex(Example, self.client)
+        obj = index._build_object(self.instance)
+        self.assertTrue(index._should_index(self.instance),
+                        "We should index an instance when should_index(instance) returns True")
+
+        instance_should_not = Example(name=None)
+        obj = index._build_object(instance_should_not)
+        self.assertFalse(index._should_index(instance_should_not),
+                        "We should not index an instance when should_index(instance) returns False")
+
+    def test_should_index_unbound(self):
+        class ExampleIndex(AlgoliaIndex):
+            fields = 'name'
+            should_index = 'static_should_index'
+        index = ExampleIndex(Example, self.client)
+        obj = index._build_object(self.instance)
+        self.assertTrue(index._should_index(self.instance),
+                        "We should index an instance when should_index() returns True")
+
+        class ExampleIndex(AlgoliaIndex):
+            fields = 'name'
+            should_index = 'static_should_not_index'
+        index = ExampleIndex(Example, self.client)
+        instance_should_not = Example()
+        obj = index._build_object(instance_should_not)
+        self.assertFalse(index._should_index(instance_should_not),
+                        "We should not index an instance when should_index() returns False")
+
+    def test_should_index_attr(self):
+        class ExampleIndex(AlgoliaIndex):
+            fields = 'name'
+            should_index = 'index_me'
+        index = ExampleIndex(Example, self.client)
+        obj = index._build_object(self.instance)
+        self.assertTrue(index._should_index(self.instance),
+                        "We should index an instance when its should_index attr is True")
+
+        instance_should_not = Example()
+        instance_should_not.index_me = False
+        obj = index._build_object(instance_should_not)
+        self.assertFalse(index._should_index(instance_should_not),
+                         "We should not index an instance when its should_index attr is False")
+
+    def test_should_index_field_raises(self):
+        class ExampleIndex(AlgoliaIndex):
+            fields = 'name'
+            should_index = 'is_admin'
+        with self.assertRaises(AlgoliaIndexError):
```

Comment:
I don't think it should raise since we want to support Django Fields. cf. my first comment: `isinstance(attr, bool)` returns `False` here. Because the `should_index` is determined during the `__init__` of the `AlgoliaIndex` instance, you don't have access to the model instance, but to the model class.